docs/usage.md

```diff
@@ -101,14 +101,14 @@ In this case, there is no `match` key, so only one rule needs to match:
 `rules` is an array of rules that can be used to target specific files with your defined syntax file.  The rules are processed until the first rule matches, so order your rules in a way that makes sense to you.
 
 ### Filename Rule
-Filename rule defines a filename with regex.  This uses the full file path.
+Filename rule defines a filename with regex.  The pattern is matched against the beginning of the full file path (there is an implicit `^`).
```

Comment:
> What i meant is that, if you were to drop the ^ from the rule you quoted, it would still behave as if it were there, because the plug-in uses re.match(), which always matches against the beginning of the string.

You are 100% correct.  I guess what I am really driving at is that both beginning and end is technically implicit.  I guess if we are going to mention the `^` we should also mention the `$`, and maybe mention that there is no adverse effect if they are used as many of the examples use them when it makes no difference (I am guilty myself).

> I suppose the reason the original author made it anchored to the beginning was so that people could type /a/b/c/ and have it work like an absolute path (but i'm not sure).

I have no problem in how he implemented it, as he did it so things could be handled differently if they are in a sub-folder with a specific name. I don't code in Ruby, but this appears to be a thing with Ruby; giving ruby files non-ruby endings:

```
{"file_name": ".*/Vagrantfile(/..*)?$"},
```

But I do feel it was named incorrectly.  If I dared change it, I would probably be flooded with issues though...

By the way, I am grateful for you taking time to help with the documentation.  I have been trying to make the documentation better, but often what is obvious to me (as I have now been in deep in the code) is often not so obvious to everyone else.  And I am not always as articulate about describing things as I think I am :smile:.